templates/common/.vscode/setup-api.ps1

```diff
@@ -0,0 +1,23 @@
+function codespacesPortUrl($portNumber) {
+    gh codespace ports `
+        -c "$CODESPACE_NAME" `
+        --json sourcePort,browseUrl `
+        --jq "map(select(.sourcePort == $portNumber))[0].browseUrl"
+}
+
+if ($env:CODESPACES -eq 'true') {
+    Write-host "Running in Codespaces. Setting port configurations."
+
+    $apiPortUrl = codespacesPortUrl 3100
+    Write-host "azd env set REACT_APP_API_BASE_URL `"$apiPortUrl`""
+    azd env set REACT_APP_API_BASE_URL "$apiPortUrl"
+
+    Write-host "Setting API port to public so web app can access it" 
+    gh codespace ports visibility 3100:public -c "$($env:CODESPACE_NAME)"
```

Comment:
Can you see what happens after you run this and then do an azd deploy?  Are the right env values set or are the new codespace values used?  We'll want to test all templates, b/c I think ACA will be diff than app service.

Reply:
Running this command and then running `azd deploy` results in the modified values being pushed to the deployment. This is not a good final state for the user. 

Couple of things we could do here: 

1. The user should run `azd env refresh` before deploying. This at least fixes the problem for now. 
1. We shuffle `.env` files around at the start of every debug session. See more on that below. 


The shuffle: 

1. Copy `.env` to `.env.bak`
3. Run `azd env set ...` to update the original `.env` file with debug-specific variables 
4. Copy the modified `.env` file to `.env.debug` 
5. The debug tasks use an `envFile` that is `${input:dotEnvFilePath}.debug` instead of `${input:dotEnvFilePath}` 
6. Add a teardown task at the end of debugging to delete the `.env.debug` file

Reply:
Testing on ACA hasn't produced problems yet. I'm seeing some strangeness on debugging func though.